### PR TITLE
Speed up Searchable Snapshot Allocator

### PR DESCRIPTION
**Pull request by original-brownbear**
This one still shows up a lot in profiling even when not using searchable
snapshots just because these settings are so costly to look up.
=> fixed the standard way by moving the setting up to index metadata

relates #77466

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java [hunk 1]
```diff
@@ -505,6 +507,10 @@ public static APIBlock readFrom(StreamInput input) throws IOException {
 
     private final AutoExpandReplicas autoExpandReplicas;
 
+    private final boolean isSearchableSnapshot;
+
+    private final boolean isPartialSearchableSnapshot;
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java:512]: NIT: Looks like there are quiet a number of fields already. Would it maker sense to merge above into enum or something?

**Review by idegtiarenko — APPROVED** [review]

File: server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java (hunk 1 above)
Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java:512]: Yea that is a fair point. I'd rather not change the logic here yet though. It's quite complicated with the two booleans in various spots and I think we should keep this kind of change as trivial as possible.
++ though to trying to find ways of reducing complexity in this class in a follow-up, I'll think about it

**Comment by original-brownbear** [comment]
Thanks Ievgen!

**Review by tlrx — APPROVED** [review]
LGTM 2, thanks for this Armin.

**Comment by original-brownbear** [comment]
Thanks Tanguy :)

**Pull request merged**